src/compiler/transformers/esDecorators.ts

```diff
@@ -0,0 +1,2087 @@
+                    const statement = factory.createVariableStatement(modifiers, varDecls);
+                    setOriginalNode(statement, node);
+                    setCommentRange(statement, getCommentRange(node));
+                    return statement;
+                }
+            }
+            else {
+                const modifiers = visitNodes(node.modifiers, node => tryCast(node, isModifier));
+                const heritageClauses = visitNodes(node.heritageClauses, visitor, isHeritageClause);
+                enterClass(/*classInfo*/ undefined);
+                const members = visitNodes(node.members, classElementVisitor, isClassElement);
+                exitClass();
+                return factory.updateClassDeclaration(node, modifiers, node.name, /*typeParameters*/ undefined, heritageClauses, members);
+            }
+        }
+
+        function visitClassExpression(node: ClassExpression, referencedName: Expression | undefined) {
+            if (isDecoratedClassLike(node)) {
+                const className = node.name ? factory.createStringLiteralFromNode(node.name) : referencedName ?? factory.createStringLiteral("");
+                const iife = transformClassLike(node, className);
+                setOriginalNode(iife, node);
+                return iife;
+            }
+            else {
+                const modifiers = visitNodes(node.modifiers, node => tryCast(node, isModifier), isModifier);
+                const heritageClauses = visitNodes(node.heritageClauses, visitor, isHeritageClause);
+                enterClass(/*classInfo*/ undefined);
+                const members = visitNodes(node.members, classElementVisitor, isClassElement);
+                exitClass();
+                return factory.updateClassExpression(node, modifiers, node.name, /*typeParameters*/ undefined, heritageClauses, members);
+            }
+        }
+
+        function prepareConstructor(parent: ClassLikeDeclaration, classInfo: ClassInfo) {
+            if (classInfo.instanceExtraInitializersName && !classInfo.hasNonAmbientInstanceFields) {
+                // If there are instance extra initializers we need to add them to the body along with any
+                // field initializers
+                const statements: Statement[] = [];
+
+                statements.push(
+                    factory.createExpressionStatement(
+                        emitHelpers().createRunInitializersHelper(
+                            factory.createThis(),
+                            classInfo.instanceExtraInitializersName
+                        )
+                    )
+                );
+
+                for (const member of parent.members) {
+                    if (isPropertyDeclaration(member) && !isStatic(member)) {
+                        // TODO(rbuckton)
+                        Debug.assertNotNode(member.name, isComputedPropertyName, "Not yet implemented.");
+
+                        let initializer = member.initializer;

```

Comment:
why does the decorator transform normal class member initialisers? Does it take that responsibility over from the classFields transform whenever there are extra instance initialisers?

Reply:
No, this is left over from prior work related to `declare` fields (postponed) and `accessor` fields (moved to `classFields` and shipping in 4.9). As I mentioned in https://github.com/microsoft/TypeScript/pull/50820#discussion_r1009741719, I will remove this.